addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,155 @@
+            return;
+        }
+
+        refreshTime = ((Number) configuration.get(REFRESH_TIME)).intValue();
+        if (refreshTime.intValue() <= 0) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
+                    DEFAULT_REFRESH_TIME);
+            refreshTime = DEFAULT_REFRESH_TIME;
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable refresher = new Runnable() {
+            @Override
+            public void run() {
+                List<Channel> channels = getThing().getChannels();
+                for (Channel channel : channels) {
+                    publishSensorValue(channel.getUID());
+                }
+            }
+        };
+
+        scheduler.scheduleWithFixedDelay(refresher, 0, refreshTime.intValue(), TimeUnit.SECONDS);
+        logger.debug("Start automatic refresh every {} seconds", refreshTime.intValue());
+    }
+
+    private void publishSensorValue(ChannelUID channelUID) {
+        String channelID = channelUID.getId();
+        switch (channelID) {
+            case TEMPERATURE:
+                publishTemperatureSensorState(channelUID);
+                break;
+            default:
+                logger.debug("Can not update channel with ID : {} - channel name might be wrong!", channelID);
+                break;
+        }
+    }
+
+    private void publishTemperatureSensorState(ChannelUID channelUID) {
+        BigDecimal temp = readSensorTemperature(gpioBusFile);
+        if (temp != null) {
+            updateState(channelUID, new DecimalType(temp));
+        }
+    }
+
+    private BigDecimal readSensorTemperature(String gpioFile) {
+        try (Stream<String> stream = Files.lines(Paths.get(gpioFile))) {
+            Optional<String> temperatureLine = stream
+                    .filter(s -> s.contains(OneWireGPIOBindingConstants.FILE_TEMP_MARKER)).findFirst();
+            if (temperatureLine.isPresent()) {
+                String line = temperatureLine.get();
+                String tempString = line.substring(line.indexOf(OneWireGPIOBindingConstants.FILE_TEMP_MARKER)
+                        + OneWireGPIOBindingConstants.FILE_TEMP_MARKER.length());
+                Integer intTemp = Integer.parseInt(tempString);
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+                return BigDecimal.valueOf(intTemp).movePointLeft(3);
+            } else {
+                logger.debug(
+                        "GPIO file didn't contain line with 't=' where temperature value should be available. Check if configuration points to the proper file");
+                return null;
+            }
+        } catch (IOException | InvalidPathException e) {
+            logger.warn("error reading GPIO bus file. File path is: {}.  Check if path is proper. {}", gpioFile, e);
```

Comment:
How can we get here? Is this a bug situation that should be reported?
If it is a user configuration error, it should rather be reported through the Thing status.
Btw, shouldn't it possibly set to OFFLINE here?